manifests/istio-telemetry/prometheus/templates/deployment.yaml

```diff
@@ -56,17 +56,220 @@ spec:
             mountPath: /etc/prometheus
           - mountPath: /etc/istio-certs
             name: istio-certs
+
+{{- if .Values.prometheus.provisionPrometheusCert }}
+        - name: istio-proxy
```

Comment:
Do we really need the entire istio-proxy config copied, just so that prometheus can take certs from the one in pilot agent? Why can't we have a simple one-off command in the pilot agent that just uses the token fetches the cert, dumps it on the specified path and then dies? (like bootstrap-cert) ? Isn't it far more simpler than dumping the entire sidecar ? This agent can run as an init container for all we care

Reply:
I agree with you but the init container won't support rotation which may be problematic. But running envoy is way overkill

Reply:
In current architecture, Istio Agent acts as SDS server and Envoy is a SDS client. The approach of having a Envoy to trigger the certificate generation and rotation is consistent with the current architecture. The pro of this approach is it requires little changes on the SDS server. The con of this approach is that it adds a Envoy to Prometheus. 
The alternative approach is to only add an Istio Agent to Prometheus and add an flow to generate and rotate certificate for Prometheus. The pro of the alternative approach is that it does not add an Envoy. The con of the alternative approach is that it requires more code changes to Istio Agent (e.g., generate, monitor, and rotate certificate), and such approach is not consistent with the current approach of SDS client triggering the certificate generation and rotation flows.